### PR TITLE
Teach `jj new` to accept multiple revisions, and make `jj merge` delegate to it

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -3477,18 +3481,22 @@ fn cmd_edit(ui: &mut Ui, command: &CommandHelper, args: &EditArgs) -> Result<(),
...
-    let parent = workspace_command.resolve_single_rev(&args.revision)?;
-    let commit_builder = CommitBuilder::for_open_commit(
-        ui.settings(),
-        parent.id().clone(),
-        parent.tree_id().clone(),
-    )
-    .set_description(args.message.clone());
+    let mut commits = vec![];
+    let mut parent_ids = vec![];
+    for revision_arg in &args.revisions {
```
Comment (yuja) [inline src/commands.rs:3481]: maybe better to assert that the revisions isn't empty. If it were empty, new commit would be created on top of the root, which seemed very wrong.

**Review by yuja — APPROVED** [review]

File: src/commands.rs (hunk 1 above)
Comment (martinvonz) [reply src/commands.rs:3481]: The default for the arguments is set to `@`, so I don't think clap ever lets this be an empty list, right?

Comment (martinvonz) [reply src/commands.rs:3481]: Oh, you're just saying we should assert that my assumption about clap holds here. Makes sense. I'll add that assertion.

Comment (martinvonz) [reply src/commands.rs:3482]: That's now done.

**Pull request merged**